Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -829,11 +829,23 @@ public class ProgramBuilder {
                 self.splice(from: program)
             }, {
                 // We can't run code generators if we don't have any visible variables.
-                guard self.scopeAnalyzer.visibleVariables.count > 0 else { return }
-                let generator = self.fuzzer.codeGenerators.randomElement()
-                if generator.requiredContext.isSubset(of: self.context) {
-                    self.run(generator)
+                if self.scopeAnalyzer.visibleVariables.isEmpty {
+                    // Generate some variables
+                    self.run(chooseUniform(from: self.fuzzer.trivialCodeGenerators))
+                    assert(!self.scopeAnalyzer.visibleVariables.isEmpty)
                 }
+                
+                // Enumerate generators that have the required context
+                var availableGenerators: [CodeGenerator] = []
+                for generator in self.fuzzer.codeGenerators {
+                    if generator.requiredContext.isSubset(of: self.context) {
+                        availableGenerators.append(generator)
+                    }
+                }
+
+                // Select a generator at random and run it
+                let generator = chooseUniform(from: availableGenerators)
```

Comment:
I think you need to handle the case that `availableGenerators` isEmpty here (and in that case, just return)

Reply:
Fair, should we also log a warning or perhaps even a assert to ensure we have generators for each context?

Reply:
So I think that this is going to be an expected situation if for example a Mutator tries to generate code right after a `BeginSwitch` but before the `BeginSwitchCase` (unless we have a CodeGenerator for just `BeginSwitchCase`). We should probably handle that case though, I guess by just aborting code generation entirely? Alternatively, I think you can also just put an assert/error in here for now, and we'll deal with this when we add restrictive contexts.

Also, while you are at it, I think you can now remove the `assert(performSplicingDuringCodeGeneration || hasVisibleVariables)` in line 825

Reply:
hmm, for now lets go with aborting code generation. We can add in assertions and testcases when we introduce restrictive contexts.